[vk-bootstrap] Build error when using x64-mingw-static

Package: vk-bootstrap[core]:x64-mingw-static -> 0.5

**Host Environment**

- Host: x64-windows
- Compiler: GNU 12.2.0
-    vcpkg-tool version: 2022-12-14-7ae0d8527fb488fde10a89c2813802dc9b03b6f9
    vcpkg-scripts version: 9067419fa 2022-12-29 (19 hours ago)

**To Reproduce**

`vcpkg install vk-bootstrap:x64-mingw-dynamic`

**Failure logs**

```
-- Downloading https://github.com/charles-lunarg/vk-bootstrap/archive/142986cdb767037118b687387b097ff6b3e7fe7d.tar.gz -> charles-lunarg-vk-bootstrap-142986cdb767037118b687387b097ff6b3e7fe7d.tar.gz...
-- Extracting source C:/Dev/C++/dngn2/vcpkg/downloads/charles-lunarg-vk-bootstrap-142986cdb767037118b687387b097ff6b3e7fe7d.tar.gz
-- Using source at C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean
-- Configuring x64-mingw-static
-- Building x64-mingw-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j17
    Working Directory: C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/x64-mingw-static-dbg
    See logs for more information:
      C:\Dev\C++\dngn2\vcpkg\buildtrees\vk-bootstrap\install-x64-mingw-static-dbg-out.log

Call Stack (most recent call first):
  C:/Dev/C++/dngn2/build/vcpkg_installed/x64-mingw-static/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  C:/Dev/C++/dngn2/build/vcpkg_installed/x64-mingw-static/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/vk-bootstrap/portfile.cmake:15 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)



```
<details><summary>C:\Dev\C++\dngn2\vcpkg\buildtrees\vk-bootstrap\install-x64-mingw-static-dbg-out.log</summary>

```
[1/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/main.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\main.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/main.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/main.cpp
FAILED: tests/CMakeFiles/vk-bootstrap-test.dir/main.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/main.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\main.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/main.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/main.cpp
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/main.cpp:2:10: fatal error: catch2/catch.hpp: No such file or directory
    2 | #include <catch2/catch.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[2/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/error_code_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\error_code_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/error_code_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/error_code_tests.cpp
FAILED: tests/CMakeFiles/vk-bootstrap-test.dir/error_code_tests.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/error_code_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\error_code_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/error_code_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/error_code_tests.cpp
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/error_code_tests.cpp:1:10: fatal error: catch2/catch.hpp: No such file or directory
    1 | #include <catch2/catch.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[3/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/unit_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\unit_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/unit_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/unit_tests.cpp
FAILED: tests/CMakeFiles/vk-bootstrap-test.dir/unit_tests.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/unit_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\unit_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/unit_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/unit_tests.cpp
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/unit_tests.cpp:1:10: fatal error: catch2/catch.hpp: No such file or directory
    1 | #include <catch2/catch.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[4/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/bootstrap_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\bootstrap_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/bootstrap_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/bootstrap_tests.cpp
FAILED: tests/CMakeFiles/vk-bootstrap-test.dir/bootstrap_tests.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT tests/CMakeFiles/vk-bootstrap-test.dir/bootstrap_tests.cpp.obj -MF tests\CMakeFiles\vk-bootstrap-test.dir\bootstrap_tests.cpp.obj.d -o tests/CMakeFiles/vk-bootstrap-test.dir/bootstrap_tests.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/bootstrap_tests.cpp
In file included from C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/bootstrap_tests.cpp:1:
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/common.h:10: warning: "NOMINMAX" redefined
   10 | #define NOMINMAX
      | 
In file included from C:/msys64/mingw64/include/c++/12.2.0/x86_64-w64-mingw32/bits/c++config.h:655,
                 from C:/msys64/mingw64/include/c++/12.2.0/bits/stl_algobase.h:59,
                 from C:/msys64/mingw64/include/c++/12.2.0/memory:63,
                 from C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/common.h:5:
C:/msys64/mingw64/include/c++/12.2.0/x86_64-w64-mingw32/bits/os_defines.h:45: note: this is the location of the previous definition
   45 | #define NOMINMAX 1
      | 
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/common.h:19:10: fatal error: GLFW/glfw3.h: No such file or directory
   19 | #include "GLFW/glfw3.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
[5/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/x64-mingw-static-dbg/example -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT example/CMakeFiles/vk-bootstrap-triangle.dir/triangle.cpp.obj -MF example\CMakeFiles\vk-bootstrap-triangle.dir\triangle.cpp.obj.d -o example/CMakeFiles/vk-bootstrap-triangle.dir/triangle.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/example/triangle.cpp
FAILED: example/CMakeFiles/vk-bootstrap-triangle.dir/triangle.cpp.obj 
C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/x64-mingw-static-dbg/example -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT example/CMakeFiles/vk-bootstrap-triangle.dir/triangle.cpp.obj -MF example\CMakeFiles\vk-bootstrap-triangle.dir\triangle.cpp.obj.d -o example/CMakeFiles/vk-bootstrap-triangle.dir/triangle.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/example/triangle.cpp
In file included from C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/example/triangle.cpp:8:
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/common.h:10: warning: "NOMINMAX" redefined
   10 | #define NOMINMAX
      | 
In file included from C:/msys64/mingw64/include/c++/12.2.0/x86_64-w64-mingw32/bits/c++config.h:655,
                 from C:/msys64/mingw64/include/c++/12.2.0/bits/stl_algobase.h:59,
                 from C:/msys64/mingw64/include/c++/12.2.0/memory:63,
                 from C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/example/triangle.cpp:3:
C:/msys64/mingw64/include/c++/12.2.0/x86_64-w64-mingw32/bits/os_defines.h:45: note: this is the location of the previous definition
   45 | #define NOMINMAX 1
      | 
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/tests/common.h:19:10: fatal error: GLFW/glfw3.h: No such file or directory
   19 | #include "GLFW/glfw3.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
[6/10] C:\msys64\mingw64\bin\x86_64-w64-mingw32-g++.exe  -IC:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src -IC:/VulkanSDK/1.3.236.0/Include -g -Wall -Wextra -Wconversion -Wsign-conversion -MD -MT CMakeFiles/vk-bootstrap.dir/src/VkBootstrap.cpp.obj -MF CMakeFiles\vk-bootstrap.dir\src\VkBootstrap.cpp.obj.d -o CMakeFiles/vk-bootstrap.dir/src/VkBootstrap.cpp.obj -c C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src/VkBootstrap.cpp
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src/VkBootstrap.cpp: In instantiation of 'void vkb::detail::VulkanFunctions::VulkanLibrary::load_func(T&, const char*) [with T = void (* (*)(VkInstance_T*, const char*))()]':
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src/VkBootstrap.cpp:76:13:   required from here
C:/Dev/C++/dngn2/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean/src/VkBootstrap.cpp:83:37: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'void (* (*)(VkInstance_T*, const char*))()' [-Wcast-function-type]
   83 |                         func_dest = reinterpret_cast<T>(GetProcAddress(library, func_name));
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "name": "dngn",
  "version-string": "0.1.0",
  "dependencies": [
    {
      "name": "sdl2",
      "features": [
        "vulkan"
      ]
    },
    {
      "name": "imgui",
      "features": [
        "sdl2-binding",
        "vulkan-binding"
      ]
    },
    "vk-bootstrap",
    "spdlog"
  ]
}

```
</details>

--- a/ports/vk-bootstrap/portfile.cmake
+++ b/ports/vk-bootstrap/portfile.cmake
@@ -10,6 +10,8 @@ vcpkg_from_github(
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DCMAKE_PROJECT_NAME=
 )
 
 vcpkg_cmake_install()
@@ -17,6 +19,6 @@ vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(COPY "${CMAKE_CURRENT_LIST_DIR}/${PORT}-config.cmake" "${CMAKE_CURRENT_LIST_DIR}/${PORT}-targets-release.cmake" "${CMAKE_CURRENT_LIST_DIR}/${PORT}-targets-debug.cmake" DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/${PORT}-config.cmake" "${CMAKE_CURRENT_LIST_DIR}/${PORT}-targets-release.cmake" "${CMAKE_CURRENT_LIST_DIR}/${PORT}-targets-debug.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 
-configure_file("${SOURCE_PATH}/LICENSE.txt" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.txt")

--- a/ports/vk-bootstrap/portfile.cmake
+++ b/ports/vk-bootstrap/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        # This option will disable build tests and example, and next release this option need to change as -DVK_BOOTSTRAP_TEST=OFF. The related upstream commit: https://github.com/charles-lunarg/vk-bootstrap/commit/4ae9513ff9182b9c519504a73435ed575a821300.
         -DCMAKE_PROJECT_NAME=
 )
 

--- a/ports/vk-bootstrap/vcpkg.json
+++ b/ports/vk-bootstrap/vcpkg.json
@@ -1,7 +1,10 @@
 {
   "name": "vk-bootstrap",
   "version": "0.5",
+  "port-version": 1,
   "description": "Vulkan bootstraping library",
+  "homepage": "https://github.com/charles-lunarg/vk-bootstrap",
+  "license": "MIT",
   "dependencies": [
     {
       "name": "vcpkg-cmake",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7986,7 +7986,7 @@
     },
     "vk-bootstrap": {
       "baseline": "0.5",
-      "port-version": 0
+      "port-version": 1
     },
     "vkfft": {
       "baseline": "1.2.17",

--- a/versions/v-/vk-bootstrap.json
+++ b/versions/v-/vk-bootstrap.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "4c2a75ef2dfe2acc1dec28e475e224bd241c9e8b",
+      "version": "0.5",
+      "port-version": 1
+    },
+    {
       "git-tree": "c882b9c4c2adf3ea8e63dae9d4ab03b4d34a4dc1",
       "version": "0.5",
       "port-version": 0

--- a/versions/v-/vk-bootstrap.json
+++ b/versions/v-/vk-bootstrap.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "4c2a75ef2dfe2acc1dec28e475e224bd241c9e8b",
+      "git-tree": "5b4e562582567c73a3bad2120a3b23e0b6b32dde",
       "version": "0.5",
       "port-version": 1
     },

